ASIN() in sql (#2582)
Codecademy/docs: content/sql/concepts/math-functions/terms/asin/asin.md

---
Title: 'ASIN()'
Description: 'Returns the arc sine of a number.'
Subjects:
  - 'Data Science'
  - 'Computer Science'
Tags:
  - 'SQLite'
  - 'PostgreSQL'
  - 'MySQL'
  - 'Functions'
CatalogContent:
  - 'learn-sql'
  - 'paths/analyze-data-with-sql'
---

> The **`ASIN()`** function in SQL is a mathematical function that returns the sine of an angle.  

Reviewer: ```suggestion
The **`ASIN()`** function in SQL is a mathematical function that returns the  inverse sine of a number.  
```

## Syntax

> ```sql
> SELECT ASIN(number);
> ```

Reviewer: ````suggestion
```pseudo
ASIN(number)
```
````
SELECT is not part of the syntax, and we generally place syntax blocks in a ` ```pseudo ` block.

> The specified number must be between -1 to 1, otherwise this function returns NULL.

Reviewer: ```suggestion
The specified number must be between -1 to 1, otherwise this function returns `NULL`.
```

## Example

In this example, the following data is given in the `numbers` table:

|       id       | input_number |
| -------------- | ------------ |
| 1              | 0.6          |
| 2              | 1            |
| 3              | -0.9         |

> The `ASIN()` function is used to calculate the arcsine as the output_number:

Reviewer: ```suggestion
The `ASIN()` function is used to calculate the arcsine as `output_number`:
```

```sql
SELECT id, input_number, ASIN(input_number) AS output_number
FROM numbers;
```

The output will be:

|       id       | input_number  |   output_number       |
| -------------- | ------------- |  ------------------   |
| 1              | 0.6           |  0.64350110879328437  | 
| 2              | 1             |  1.5707963267948966   |
| 3              | -0.9          |  -1.1197695149986342  |

> It works with with SQL Server (starting with 2008), Azure SQL Database, Azure SQL Data Warehouse, Parallel Data Warehouse
> 

Reviewer: ```suggestion
```
Delete this line as it is somewhat redundant with the note.
